WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/DiscountLineDetailsView.swift

```diff
@@ -0,0 +1,136 @@
+import SwiftUI
+
+struct DiscountLineDetailsView: View {
+
+    @ObservedObject private var viewModel: FeeOrDiscountLineDetailsViewModel
+
+    init(viewModel: FeeOrDiscountLineDetailsViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: .zero) {
+            Text(Localization.fixedPriceDiscountLabel)
+                .renderedIf(viewModel.feeOrDiscountType == .fixed)
+                .padding()
+            Text(Localization.percentagePriceDiscountLabel)
+                .renderedIf(viewModel.feeOrDiscountType == .percentage)
+                .padding()
+            HStack {
+                inputFixedField
+                    .renderedIf(viewModel.feeOrDiscountType == .fixed)
+                inputPercentageField
+                    .renderedIf(viewModel.feeOrDiscountType == .percentage)
+                HStack(spacing: 0) {
+                    Button(viewModel.currencySymbol) {
+                        viewModel.feeOrDiscountType = .fixed
+                    }
+                    .buttonStyle(isPrimary: viewModel.feeOrDiscountType == .fixed)
+                    .fixedSize(horizontal: true, vertical: false)
+                    Button(viewModel.percentSymbol) {
+                        viewModel.feeOrDiscountType = .percentage
+                    }
+                    .buttonStyle(isPrimary: viewModel.feeOrDiscountType == .percentage)
+                    .fixedSize(horizontal: true, vertical: false)
+                }
+            }
+        }
```

Comment:
There's a lot of checks for `.fixed` or `.percentage` here. Have you tried making two separate views, one for each, and switching between them with a top level `if`? It would probably be easier to read, and I _think_ the transitions should still work correctly... but they might not.

Reply:
I agree there's room for improvement here. I might also be able to simplify further once we get rid of the "fees" part of the model. Logged on https://github.com/woocommerce/woocommerce-ios/issues/10915